curriculum/challenges/english/blocks/lecture-working-with-npm-scripts/692aba0978ca391f10455b4a.md

```diff
@@ -0,0 +1,327 @@
+
+If you have worked with Git, you may think that this file sounds very similar to `.gitignore`, right? Well, that's because it is. According to the [npm documentation](https://docs.npmjs.com/cli/v9/using-npm/developers), "If there's no `.npmignore` file, but there is a `.gitignore` file, then npm will ignore the stuff matched by the `.gitignore` file."
+
+You should also know that npm looks for `.npmignore` and `.gitignore` files in all subdirectories of your package, not just the root directory.
+
+By default, these paths are automatically ignored, so you don't need to add them to `.npmignore`:
+
+* `.*.swp`
+    
+* `._*`
+    
+* `.DS_Store`
+    
+* `.git`
+    
+* `.gitignore`
+    
+* `.hg`
+    
+* `.npmignore`
+    
+* `.npmrc`
+    
+* `.lock-wscript`
+    
+* `.svn`
+    
+* `.wafpickle-*`
+    
+* `config.gypi`
+    
+* `CVS`
+    
+* `npm-debug.log`
+    
+
+The `node_modules` directory is also ignored, except bundled dependencies.
+
+## **Step 6: Publish your package**
+
+Before publishing your package to the npm registry, you should test it by running this command:
+
+```bash
+npm install <full-path-to-package-directory>
+```
+
+Once you are ready to publish it, the command will depend on the type of package: scope vs. unscoped.
+
+If your package is unscoped, run this command:
+
+```bash
+npm publish
+```
+
+If your package is scoped, it will be published as a private package by default (if this feature is available in your account). If you want to make it public, you will need to add `--access public` to the command.
+
+```bash
+npm publish --access public
+```
+
+To see your public package, go to this URL:
+
+```bash
+https://npmjs.com/package/<your-package-name>
+```
+
+You should see an indication of the visibility of your package (public or private) below the package name.
+
+## **Step 7 (Optional): Install Your Package**
+
+Now that you are all set up, you can install your package in a Node.js project with this command:
+
+```bash
+npm install <your-package-name>
+```
+
+## Updating Your Package
+
+If you need to make changes to your package after publishing it, you can update the package version in `package.json`.
+
+This is the [standard convention](https://docs.npmjs.com/about-semantic-versioning#incrementing-semantic-versions-in-published-packages) for semantic versioning:
+
+* **First release:** 1.0.0.
+    
+* **Patch release:** Increment the third digit. For example, updating from 1.0.0 to 1.0.1.
+    
+* **Minor release:** Increment the middle digit and reset the last digit to zero. For example, updating 1.0.0 to 1.1.0.
+    
+* **Major release:** Increment the first digit and reset the middle and last digits to zero. For example, updating 1.1.0 to 2.0.0.
```

Comment:
Maybe put all the versions here in backticks and quotes? Probably fine as-is if we want.